docs/redis.md

```diff
@@ -0,0 +1,34 @@
+# Usage of Redis for cozy-stack
+
+
+It will be important to figure out if we use a cluster of Redis or only one.
+Assuming a cluster allows us to scale better in the future but for now we will focus on a single redis target for the stack.
+
+## As a cache / store (KV)
+
+Already implemented in https://github.com/cozy/cozy-stack/blob/master/pkg/cache/cache.go
+
+To use with sessions / download manager
+
+
+## As a simple lock
+
+Usage ?
+
+ref
+https://redis.io/topics/distlock
+(algorithm for cluster)
+
+
+## As a FS lock (range+RW)
+
+Using a Range+RW lock is an optimization, it would be acceptable to use a simple lock instead to start.
```

Comment:
- Range: I don't want of this optimization. It looks like what we can gain doesn't worth its complexity. Except, of course, if a benchmark shows that I'm wrong. But let's forget it for now.

- RW: it looks like it's complicated to implement, but I think it has the potential to really improve the performances (mainly by allowing several readers at the same time). Let's keep it for later.